required/latex-lab/latex-lab-everymath.dtx

```diff
@@ -0,0 +1,347 @@
+% \iffalse meta-comment
+%
+%% File: latex-lab-everymath.dtx
+% Copyright (C) 2022 The LaTeX Project
+%
+% It may be distributed and/or modified under the conditions of the
+% LaTeX Project Public License (LPPL), either version 1.3c of this
+% license or (at your option) any later version.  The latest version
+% of this license is in the file
+%
+%    https://www.latex-project.org/lppl.txt
+%
+%
+% The development version of the bundle can be found below
+%
+%    https://github.com/latex3/latex2e/required/latex-lab
+%
+% for those people who are interested or want to report an issue.
+%
+%<*driver>
+\documentclass{l3doc}
+\EnableCrossrefs
+\CodelineIndex
+\begin{document}
+  \DocInput{latex-lab-everymath.dtx}
+\end{document}
+%</driver>
+%
+% \fi
+%
+%
+% \title{The \texttt{latex-lab-everymath} code\thanks{}}
+% \author{\LaTeX{} Project}
+%
+% \maketitle
+%
+% \newcommand\fmi[1]{\begin{quote} TODO: \itshape #1\end{quote}}
+% \newcommand\NEW[1]{\marginpar{\mbox{}\hfill\fbox{New: #1}}}
+% \providecommand\class[1]{\texttt{#1.cls}}
+% \providecommand\pkg[1]{\texttt{#1}}
+% \providecommand\hook[1]{\texttt{#1}}
+%
+% \begin{abstract}
+% \end{abstract}
+%
+% \tableofcontents
+%
+% This file implements capture of all math mode material at the top level.
+% It provides one code-level interface to use this captured input.
+%
+% \begin{function}{\math_processor:n}
+%   \begin{syntax}
+%     \cs{math_processor:n} \Arg{tokens}
+%   \end{syntax}
+%   Declares that the captured math content should be passed to the
+%   \meta{tokens}, which will receive the content as |#1|.
+% \end{function}
+%
+% \MaybeStop{\setlength\IndexMin{200pt}  \PrintIndex  }
+%
+% \section{The Implementation}
+%
+%    \begin{macrocode}
+%<@@=math>
+%    \end{macrocode}
+%
+%    \begin{macrocode}
+%<*kernel>
+%    \end{macrocode}
+%
+% \subsection{File declaration}
+%    \begin{macrocode}
+\ProvidesFile{latex-lab-everymath.ltx}
+        [2022-12-09 v0.1a Grab all the math(s)]
+%    \end{macrocode}
+%
+%    \begin{macrocode}
+\ExplSyntaxOn
+%    \end{macrocode}
+%
+% \subsection{Setup}
+%
+% Loading \pkg{amsmath} is an absolute requirement: this avoids needing to
+% have conditional definitions and deals with how to define \cs{[}/\cs{]}
+% neatly.
+%    \begin{macrocode}
+\AddToHook { begindocument / before }
+  { \RequirePackage { amsmath } }
+%    \end{macrocode} 
```

Comment:
I think this would be better implemented by having a latex-lab-requirements.dtx file that hold everything we want to automatically load, possibly based on keys in \DocumentMetadata. That might need some thoughs/design, but I think it should be centralized.